From c004abc0d5280b2377a6df2407bb647ae62c5d10 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20D=C3=B6rr?= <timo@latecrew.de>
Date: Tue, 15 May 2012 21:07:38 +0200
Subject: [PATCH] Switch: ige-mac-integration to gtk-mac-integration

-Updated DllImport references to new gtkmacintegration libs
-Updated the ige_* symbols to gtk_* symbols
-still using the old (deprecated) Carbon-based gtk-mac-integration API
instead of the proposed GtkOSXApplication API (which resides in the same
library)
---
 build/build.rules.mk                               |    2 +-
 build/m4/banshee/osx.m4                            |    2 +-
 .../Banshee.Osx/OsxIntegration.Ige/IgeMacMenu.cs   |   30 ++++++++++----------
 .../OsxIntegration.Ige/IgeMacMenuGroup.cs          |    6 ++--
 4 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/build/build.rules.mk b/build/build.rules.mk
index c387972..344b90f 100644
--- a/build/build.rules.mk
+++ b/build/build.rules.mk
@@ -39,7 +39,7 @@ $(ASSEMBLY_FILE): $(SOURCES_BUILD) $(RESOURCES_EXPANDED) $(DEP_LINK)
 	@mkdir -p $(top_builddir)/bin
 	@if [ ! "x$(ENABLE_RELEASE)" = "xyes" ]; then \
 		$(top_srcdir)/build/dll-map-makefile-verifier $(srcdir)/Makefile.am $(srcdir)/$(notdir $@.config) && \
-		$(MONO) $(top_builddir)/build/dll-map-verifier.exe $(srcdir)/$(notdir $@.config) -iwinmm -ilibbanshee -ilibbnpx11 -ilibc -ilibc.so.6 -iintl -ilibmtp.dll -ilibigemacintegration.dylib -iCFRelease $(SOURCES_BUILD); \
+		$(MONO) $(top_builddir)/build/dll-map-verifier.exe $(srcdir)/$(notdir $@.config) -iwinmm -ilibbanshee -ilibbnpx11 -ilibc -ilibc.so.6 -iintl -ilibmtp.dll -ilibgtkmacintegration.dylib -iCFRelease $(SOURCES_BUILD); \
 	fi;
 	$(MCS) \
 		$(GMCS_FLAGS) \
diff --git a/build/m4/banshee/osx.m4 b/build/m4/banshee/osx.m4
index bc841f3..4eee7cc 100644
--- a/build/m4/banshee/osx.m4
+++ b/build/m4/banshee/osx.m4
@@ -3,7 +3,7 @@ AC_DEFUN([BANSHEE_CHECK_OSX],
 	enable_osx="no"
 	if test "x${host_os%${host_os#??????}}" = "xdarwin"; then
 		enable_osx="yes"
-		PKG_CHECK_MODULES(IGEMACINTEGRATION, ige-mac-integration >= 0.8.6)
+		PKG_CHECK_MODULES(GTKMACINTEGRATION, gtk-mac-integration >= 1.0.1)
 		PKG_CHECK_MODULES(MONOMAC, monomac >= 0.7)
 		MONOMAC_ASSEMBLIES=`$PKG_CONFIG --variable=Libraries monomac`
 		AC_SUBST(MONOMAC_LIBS)
diff --git a/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenu.cs b/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenu.cs
index f34d4fc..aea7f5f 100644
--- a/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenu.cs
+++ b/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenu.cs
@@ -31,41 +31,41 @@ namespace OsxIntegration.Ige
 {
     public static class IgeMacMenu
     {
-        [DllImport ("libigemacintegration.dylib")]
-        private static extern void ige_mac_menu_connect_window_key_handler (IntPtr window);
+        [DllImport ("libgtkmacintegration.dylib")]
+        private static extern void gtk_mac_menu_connect_window_key_handler (IntPtr window);
 
         public static void ConnectWindowKeyHandler (Gtk.Window window)
         {
-            ige_mac_menu_connect_window_key_handler (window.Handle);
+            gtk_mac_menu_connect_window_key_handler (window.Handle);
         }
 
-        [DllImport ("libigemacintegration.dylib")]
-        private static extern void ige_mac_menu_set_global_key_handler_enabled (bool enabled);
+        [DllImport ("libgtkmacintegration.dylib")]
+        private static extern void gtk_mac_menu_set_global_key_handler_enabled (bool enabled);
 
         public static bool GlobalKeyHandlerEnabled {
-            set { ige_mac_menu_set_global_key_handler_enabled (value); }
+            set { gtk_mac_menu_set_global_key_handler_enabled (value); }
         }
 
-        [DllImport ("libigemacintegration.dylib")]
-        static extern void ige_mac_menu_set_menu_bar (IntPtr menu_shell);
+        [DllImport ("libgtkmacintegration.dylib")]
+        static extern void gtk_mac_menu_set_menu_bar (IntPtr menu_shell);
 
         public static Gtk.MenuShell MenuBar {
-            set { ige_mac_menu_set_menu_bar (value == null ? IntPtr.Zero : value.Handle); }
+            set { gtk_mac_menu_set_menu_bar (value == null ? IntPtr.Zero : value.Handle); }
         }
 
-        [DllImport ("libigemacintegration.dylib")]
-        private static extern void ige_mac_menu_set_quit_menu_item (IntPtr quit_item);
+        [DllImport ("libgtkmacintegration.dylib")]
+        private static extern void gtk_mac_menu_set_quit_menu_item (IntPtr quit_item);
 
         public static Gtk.MenuItem QuitMenuItem {
-            set { ige_mac_menu_set_quit_menu_item (value == null ? IntPtr.Zero : value.Handle); }
+            set { gtk_mac_menu_set_quit_menu_item (value == null ? IntPtr.Zero : value.Handle); }
         }
 
-        [DllImport ("libigemacintegration.dylib")]
-        private static extern IntPtr ige_mac_menu_add_app_menu_group ();
+        [DllImport ("libgtkmacintegration.dylib")]
+        private static extern IntPtr gtk_mac_menu_add_app_menu_group ();
 
         public static IgeMacMenuGroup AddAppMenuGroup ()
         {
-            var native = ige_mac_menu_add_app_menu_group ();
+            var native = gtk_mac_menu_add_app_menu_group ();
             return native == IntPtr.Zero
                 ? null
                 : (IgeMacMenuGroup)GLib.Opaque.GetOpaque (native, typeof (IgeMacMenuGroup), false);
diff --git a/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenuGroup.cs b/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenuGroup.cs
index 57ed82a..e8160cb 100644
--- a/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenuGroup.cs
+++ b/src/Backends/Banshee.Osx/OsxIntegration.Ige/IgeMacMenuGroup.cs
@@ -35,15 +35,15 @@ namespace OsxIntegration.Ige
         {
         }
 
-        [DllImport ("libigemacintegration.dylib")]
-        private static extern void ige_mac_menu_add_app_menu_item (IntPtr raw,
+        [DllImport ("libgtkmacintegration.dylib")]
+        private static extern void gtk_mac_menu_add_app_menu_item (IntPtr raw,
             IntPtr menu_item, IntPtr label);
 
         public void AddMenuItem (Gtk.MenuItem menu_item, string label)
         {
             var native_label = GLib.Marshaller.StringToPtrGStrdup (label);
             try {
-                ige_mac_menu_add_app_menu_item (Handle,
+                gtk_mac_menu_add_app_menu_item (Handle,
                     menu_item == null ? IntPtr.Zero : menu_item.Handle,
                     native_label);
             } finally {
-- 
1.7.7.5 (Apple Git-26)

